src/controllers/users.go

```diff
@@ -0,0 +1,169 @@
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could fetch rolebinding %q with error: %v", userIdentifier.UserName, err.Error()))
+		return userOutput, err
+	}
+	u.logger.Debug(fmt.Sprintf("fetched roleBinding %q successfully", userIdentifier.UserName))
+	userOutput.Name = roleBinding.Name
+	userOutput.Role = convertToPlatformRole(roleBinding.RoleRef.Name)
+
+	return userOutput, nil
+}
+
+func (u *userController) AddUser(user types.UserInput) (types.User, error) {
+	userOutput := types.User{}
+	u.logger.Debug(fmt.Sprintf("Trying to create rolebinding %q in %q namespace", user.Name, user.Namespace))
+
+	roleBinding, err := u.client.RbacV1().RoleBindings(user.Namespace).Create(u.ctx,
+		prepareRoleBinding(user.Name, user.Role), metav1.CreateOptions{})
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could create rolebinding %q with error: %v", user.Name, err.Error()))
+		return userOutput, err
+	}
+	u.logger.Debug(fmt.Sprintf("created roleBinding %q successfully", user.Name))
+	userOutput.Name = roleBinding.Name
+	userOutput.Role = user.Role
+
+	return userOutput, nil
+}
+
+func (u *userController) UpdateUser(user types.UserInput) (types.User, error) {
+	userOutput := types.User{}
+
+	u.logger.Debug(fmt.Sprintf("Trying to update rolebinding %q in %q namespace", user.Name, user.Namespace))
+
+	_, err := u.DeleteUser(types.UserIdentifier{UserName: user.Name, NamespaceName: user.Namespace})
+	if err != nil {
+		return userOutput, err
+	}
+
+	err = retry.OnError(retry.DefaultRetry, func(err error) bool {
+		return err != nil
+
+	}, func() error {
+		_, err := u.AddUser(user)
+		return err
+	})
+
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could not update rolebinding %q with error: %v", user.Name, err.Error()))
+		return userOutput, err
+	}
+
+	u.logger.Debug(fmt.Sprintf("updated roleBinding %q successfully", user.Name))
+	userOutput.Name = user.Name
+	userOutput.Role = user.Role
+
+	return userOutput, nil
+}
+
+func (u *userController) DeleteUser(userIdentifier types.UserIdentifier) (types.DeleteUserResponse, error) {
+	u.logger.Debug(fmt.Sprintf("Trying to delete rolebinding %q in %q namespace", userIdentifier.UserName, userIdentifier.NamespaceName))
+
+	if err := u.client.RbacV1().RoleBindings(userIdentifier.NamespaceName).Delete(u.ctx, userIdentifier.UserName, metav1.DeleteOptions{}); err != nil {
+		u.logger.Error(fmt.Sprintf("Could note delete rolebinding %q with error: %v",
+			userIdentifier.UserName, err.Error()))
+		return types.DeleteUserResponse{Message: fmt.Sprintf("Could note delete rolebinding %q with error: %s",
+			userIdentifier.UserName, err.Error())}, err
+	}
+
+	u.logger.Debug(fmt.Sprintf("deleted roleBinding %q successfully", userIdentifier.UserName))
+	return types.DeleteUserResponse{Message: fmt.Sprintf("deleted roleBinding %q successfully", userIdentifier.UserName)}, nil
+}
+
+func prepareRoleBinding(roleBindingName string, role string) *rbacv1.RoleBinding {
```

Comment:
nit: I'd call it `newRoleBinding` as it returns a new `rbacv1.RoleBinding`.

Reply:
Its conventions for us on the other operators we develop